lib/query.js

```diff
@@ -277,12 +277,14 @@ module.exports = class Query extends Readable {
     }
   }
 
-  _onerror (_, req) {
+  _onerror (err, req) {
     const addr = req.to.host + ':' + req.to.port
     const refs = this._seen.get(addr)
 
-    this._seen.set(addr, DOWN)
-    for (const node of refs) this._downHint(node, req.to)
+    if (err?.code === 'REQUEST_TIMEOUT') {
```

Comment:
Note: can change to `err.code` if err is guaranteed to be an object

Reply:
Error always set and always an error yes